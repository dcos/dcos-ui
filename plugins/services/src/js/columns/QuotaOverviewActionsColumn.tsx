import * as React from "react";
import { Cell, Icon } from "@dcos/ui-kit";
import { SystemIcons } from "@dcos/ui-kit/dist/packages/icons/dist/system-icons-enum";
import {
  greyDark,
  iconSizeXs,
} from "@dcos/ui-kit/dist/packages/design-tokens/build/js/designTokens";

import { ServiceGroup } from "#PLUGINS/services/src/js/types/ServiceGroup";

export function actionsRenderer(_group: ServiceGroup) {
  return (
    <Cell>
      <span className="icon-margin-right">
        <Icon
          color={greyDark}
          shape={SystemIcons.EllipsisVertical}
          size={iconSizeXs}
        />
      </span>
    </Cell>
  );
}
