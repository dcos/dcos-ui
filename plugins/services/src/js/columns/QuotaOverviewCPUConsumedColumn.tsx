import * as React from "react";
import { Trans } from "@lingui/macro";
import { Cell, Tooltip } from "@dcos/ui-kit";

import { ServiceGroup } from "#PLUGINS/services/src/js/types/ServiceGroup";
import ProgressBar from "#SRC/js/components/ProgressBar";
import * as ResourcesUtil from "#SRC/js/utils/ResourcesUtil";

const className = `color-${ResourcesUtil.getResourceColor("cpus")}`;

export function cpuRenderer(group: ServiceGroup) {
  const quota = ServiceGroup.getQuota(group, "cpus");
  return (
    <Cell>
      {quota && quota.limit ? (
        <div>
          <ProgressBar
            data={ProgressBar.getDataFromValue(
              ServiceGroup.getQuotaPercentage(group, "cpus"),
              className
            )}
            total={100}
          />
          <div className="table-content-spacing-left table-content--inline-block">
            <Tooltip
              id="quota-cpu-tooltip"
              trigger={
                <span>{ServiceGroup.getQuotaPercentage(group, "cpus")}%</span>
              }
            >
              <Trans render="span">
                {quota.consumed} of {quota.limit} Cores
              </Trans>
            </Tooltip>
          </div>
        </div>
      ) : (
        <div>
          <Trans>No Limit</Trans>
        </div>
      )}
    </Cell>
  );
}
