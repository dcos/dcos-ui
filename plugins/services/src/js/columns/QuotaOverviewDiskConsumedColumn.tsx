import * as React from "react";
import { Trans } from "@lingui/macro";
import { Cell, Tooltip } from "@dcos/ui-kit";

import { ServiceGroup } from "#PLUGINS/services/src/js/types/ServiceGroup";
import ProgressBar from "#SRC/js/components/ProgressBar";
import * as ResourcesUtil from "#SRC/js/utils/ResourcesUtil";

const className = `color-${ResourcesUtil.getResourceColor("disk")}`;

export function diskRenderer(group: ServiceGroup) {
  const quota = ServiceGroup.getQuota(group, "disk");
  return (
    <Cell>
      {quota && quota.limit ? (
        <div>
          <ProgressBar
            data={ProgressBar.getDataFromValue(
              ServiceGroup.getQuotaPercentage(group, "disk"),
              className
            )}
            total={100}
          />
          <div className="table-content-spacing-left">
            <Tooltip
              id="quota-disk-tooltip"
              trigger={
                <span>{ServiceGroup.getQuotaPercentage(group, "disk")}%</span>
              }
            >
              <Trans render="span">
                {quota.consumed} of {quota.limit} MiB
              </Trans>
            </Tooltip>
          </div>
        </div>
      ) : (
        <div>
          <Trans>No Limit</Trans>
        </div>
      )}
    </Cell>
  );
}
