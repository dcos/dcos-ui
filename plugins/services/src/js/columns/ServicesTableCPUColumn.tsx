import * as React from "react";
import { NumberCell } from "@dcos/ui-kit";

import Pod from "../structs/Pod";
import Service from "../structs/Service";
import ServiceTree from "../structs/ServiceTree";
import Units from "#SRC/js/utils/Units";
import { SortDirection } from "plugins/services/src/js/types/SortDirection";
import ServiceTableUtil from "../utils/ServiceTableUtil";

export function cpuRenderer(
  service: Service | Pod | ServiceTree
): React.ReactNode {
  const resource = service.getResources()[`cpus`];

  return (
    <NumberCell>
      <span>{Units.formatResource("cpus", resource)}</span>
    </NumberCell>
  );
}

export function cpuSorter(
  data: Array<Service | Pod | ServiceTree>,
  sortDirection: SortDirection
): any {
  return ServiceTableUtil.sortData(data, sortDirection, "cpus");
}
