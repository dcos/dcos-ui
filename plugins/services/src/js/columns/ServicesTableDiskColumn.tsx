import * as React from "react";
import { NumberCell } from "@dcos/ui-kit";

import Pod from "../structs/Pod";
import Service from "../structs/Service";
import ServiceTree from "../structs/ServiceTree";
import Units from "#SRC/js/utils/Units";
import { SortDirection } from "plugins/services/src/js/types/SortDirection";
import ServiceTableUtil from "../utils/ServiceTableUtil";

export function diskRenderer(
  service: Service | Pod | ServiceTree
): React.ReactNode {
  const resource = service.getResources()[`disk`];

  return (
    <NumberCell>
      <span>{Units.formatResource("disk", resource)}</span>
    </NumberCell>
  );
}

export function diskSorter(
  data: Array<Service | Pod | ServiceTree>,
  sortDirection: SortDirection
): any {
  return ServiceTableUtil.sortData(data, sortDirection, "disk");
}
