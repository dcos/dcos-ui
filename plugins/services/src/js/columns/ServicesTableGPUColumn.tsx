import * as React from "react";
import { NumberCell } from "@dcos/ui-kit";

import Pod from "../structs/Pod";
import Service from "../structs/Service";
import ServiceTree from "../structs/ServiceTree";
import Units from "#SRC/js/utils/Units";
import { SortDirection } from "plugins/services/src/js/types/SortDirection";
import ServiceTableUtil from "../utils/ServiceTableUtil";

export function gpuRenderer(
  service: Service | Pod | ServiceTree
): React.ReactNode {
  const resource = service.getResources()[`gpus`];

  return (
    <NumberCell>
      <span>{Units.formatResource("gpus", resource)}</span>
    </NumberCell>
  );
}

export function gpuSorter(
  data: Array<Service | Pod | ServiceTree>,
  sortDirection: SortDirection
): any {
  return ServiceTableUtil.sortData(data, sortDirection, "gpus");
}
