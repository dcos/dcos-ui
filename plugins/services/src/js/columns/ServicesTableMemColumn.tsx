import * as React from "react";
import { NumberCell } from "@dcos/ui-kit";

import Pod from "../structs/Pod";
import Service from "../structs/Service";
import ServiceTree from "../structs/ServiceTree";
import Units from "#SRC/js/utils/Units";
import { SortDirection } from "plugins/services/src/js/types/SortDirection";
import ServiceTableUtil from "../utils/ServiceTableUtil";

export function memRenderer(
  service: Service | Pod | ServiceTree
): React.ReactNode {
  const resource = service.getResources()[`mem`];
  return (
    <NumberCell>
      <span>{Units.formatResource("mem", resource)}</span>
    </NumberCell>
  );
}

export function memSorter(
  data: Array<Service | Pod | ServiceTree>,
  sortDirection: SortDirection
): any {
  return ServiceTableUtil.sortData(data, sortDirection, "mem");
}
