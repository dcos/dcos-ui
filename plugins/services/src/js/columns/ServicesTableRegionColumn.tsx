import * as React from "react";
import { TextCell } from "@dcos/ui-kit";
import { Tooltip } from "reactjs-components";

import CompositeState from "#SRC/js/structs/CompositeState";
import Pod from "../structs/Pod";
import Service from "../structs/Service";
import ServiceTree from "../structs/ServiceTree";
import { SortDirection } from "plugins/services/src/js/types/SortDirection";
import ServiceTableUtil from "../utils/ServiceTableUtil";

export function regionRenderer(
  service: Service | Pod | ServiceTree
): React.ReactNode {
  const localRegion: string = CompositeState.getMasterNode().getRegionName();
  let regions: string[] = service.getRegions();

  regions = regions.map(region =>
    region === localRegion ? region + " (Local)" : region
  );

  if (regions.length === 0) {
    regions.push("N/A");
  }

  return (
    <TextCell>
      <Tooltip elementTag="span" wrapText={true} content={regions.join(", ")}>
        {regions.join(", ")}
      </Tooltip>
    </TextCell>
  );
}

export function regionSorter(
  data: Array<Service | Pod | ServiceTree>,
  sortDirection: SortDirection
): any {
  return ServiceTableUtil.sortData(data, sortDirection, "region");
}
