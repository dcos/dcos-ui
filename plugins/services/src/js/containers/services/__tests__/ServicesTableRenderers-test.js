import { NumberCell } from "@dcos/ui-kit";

import { regionRenderer } from "../../../columns/ServicesTableRegionColumn";
import { cpuRenderer } from "../../../columns/ServicesTableCPUColumn";
import { memRenderer } from "../../../columns/ServicesTableMemColumn";
import { diskRenderer } from "../../../columns/ServicesTableDiskColumn";
import { gpuRenderer } from "../../../columns/ServicesTableGPUColumn";

// Explicitly mock for react-intl
jest.mock("../../../components/modals/ServiceActionDisabledModal");

/* eslint-disable no-unused-vars */
const React = require("react");
/* eslint-enable no-unused-vars */
const renderer = require("react-test-renderer");
const Application = require("../../../structs/Application");

function renderNumberCell(value) {
  return (
    <NumberCell>
      <span>{value}</span>
    </NumberCell>
  );
}

describe("ServicesTable", function() {
  const healthyService = new Application({
    healthChecks: [{ path: "", protocol: "HTTP" }],
    cpus: 1,
    gpus: 1,
    instances: 1,
    mem: 2048,
    disk: 0,
    tasksStaged: 0,
    tasksRunning: 2,
    tasksHealthy: 2,
    tasksUnhealthy: 0
  });

  describe("#renderStats", function() {
    it("renders resources/stats cpus property", function() {
      var cpusCell = cpuRenderer(healthyService);

      expect(cpusCell).toEqual(renderNumberCell(1));
    });

    it("renders resources/stats gpus property", function() {
      var gpusCell = gpuRenderer(healthyService);

      expect(gpusCell).toEqual(renderNumberCell(1));
    });

    it("renders resources/stats mem property", function() {
      var memCell = memRenderer(healthyService);

      expect(memCell).toEqual(renderNumberCell("2 GiB"));
    });

    it("renders resources/stats disk property", function() {
      var disksCell = diskRenderer(healthyService);

      expect(disksCell).toEqual(renderNumberCell("0 B"));
    });

    it("renders sum of resources/stats cpus property", function() {
      const application = new Application({
        healthChecks: [{ path: "", protocol: "HTTP" }],
        cpus: 1,
        instances: 2,
        mem: 2048,
        disk: 0,
        tasksStaged: 0,
        tasksRunning: 2,
        tasksHealthy: 2,
        tasksUnhealthy: 0,
        getResources: jest.fn()
      });

      application.getResources.mockReturnValue({ cpus: application.cpus });
      var cpusCell = cpuRenderer(application);

      expect(cpusCell).toEqual(renderNumberCell(1));
    });

    it("renders sum of resources/stats gpus property", function() {
      const application = new Application({
        healthChecks: [{ path: "", protocol: "HTTP" }],
        cpus: 1,
        gpus: 1,
        instances: 2,
        mem: 2048,
        disk: 0,
        tasksStaged: 0,
        tasksRunning: 2,
        tasksHealthy: 2,
        tasksUnhealthy: 0,
        getResources: jest.fn()
      });

      application.getResources.mockReturnValue({ gpus: application.gpus });
      var gpusCell = gpuRenderer(application);

      expect(gpusCell).toEqual(renderNumberCell(application.gpus));
    });

    it("renders sum of resources/stats mem property", function() {
      const application = new Application({
        healthChecks: [{ path: "", protocol: "HTTP" }],
        cpus: 1,
        gpus: 1,
        instances: 2,
        mem: 2048,
        disk: 0,
        tasksStaged: 0,
        tasksRunning: 2,
        tasksHealthy: 2,
        tasksUnhealthy: 0,
        getResources: jest.fn()
      });

      application.getResources.mockReturnValue({ mem: application.mem });
      var memCell = memRenderer(application);

      expect(memCell).toEqual(renderNumberCell("2 GiB"));
    });

    it("renders sum of resources/stats disk property", function() {
      const application = new Application({
        healthChecks: [{ path: "", protocol: "HTTP" }],
        cpus: 1,
        gpus: 1,
        instances: 2,
        mem: 2048,
        disk: 0,
        tasksStaged: 0,
        tasksRunning: 2,
        tasksHealthy: 2,
        tasksUnhealthy: 0,
        getResources: jest.fn()
      });

      application.getResources.mockReturnValue({ disk: application.disk });
      var diskCell = diskRenderer(application);

      expect(diskCell).toEqual(renderNumberCell("0 B"));
    });
  });

  describe("#renderRegions", function() {
    const renderRegions = regionRenderer;
    let mockService;
    beforeEach(function() {
      mockService = {
        getRegions: jest.fn()
      };
    });

    it("renders with no regions", function() {
      mockService.getRegions.mockReturnValue([]);

      expect(
        renderer.create(renderRegions(mockService)).toJSON()
      ).toMatchSnapshot();
    });

    it("renders with one region", function() {
      mockService.getRegions.mockReturnValue(["aws/eu-central-1"]);

      expect(
        renderer.create(renderRegions(mockService)).toJSON()
      ).toMatchSnapshot();
    });

    it("renders with multiple regions", function() {
      mockService.getRegions.mockReturnValue(["aws/eu-central-1", "dc-east"]);

      expect(
        renderer.create(renderRegions(mockService)).toJSON()
      ).toMatchSnapshot();
    });
  });
});
