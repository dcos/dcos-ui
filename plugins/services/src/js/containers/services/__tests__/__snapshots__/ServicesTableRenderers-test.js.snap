// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesTable #renderRegions renders with multiple regions 1`] = `
<div
  className="css-1c5c0a1"
>
  <span
    className="tooltip-wrapper text-align-center"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    aws/eu-central-1, dc-east
  </span>
</div>
`;

exports[`ServicesTable #renderRegions renders with no regions 1`] = `
<div
  className="css-1c5c0a1"
>
  <span
    className="tooltip-wrapper text-align-center"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    N/A
  </span>
</div>
`;

exports[`ServicesTable #renderRegions renders with one region 1`] = `
<div
  className="css-1c5c0a1"
>
  <span
    className="tooltip-wrapper text-align-center"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    aws/eu-central-1
  </span>
</div>
`;
